src/components/button.js

```diff
@@ -0,0 +1,121 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+import { ButtonBase } from '../base-components'
+import clsx from 'clsx'
+
+
+const buttonConfigs = () => ({
+  colors: {
```

Comment:
I think for color it is better to be as dynamic as possible. I can't imagine having a whole palette inside of button.js in buttonConfigs

Reply:
sorry for this massive button configs object, i'll try to see if i could reduce it a bit
but this isn't the color palette, the color palette is in `tailwind.config.js` that looks something like this:
```
      'secondary-50': '#fafafa',
      'secondary-100': '#f5f5f5',
      'secondary-200': '#eeeeee',
      'secondary-300': '#e0e0e0',
      'secondary-400': '#bdbdbd',
      'secondary-500': '#9e9e9e',
      'secondary-600': '#757575',
      'secondary-700': '#616161',
      'secondary-800': '#424242',
      'secondary-900': '#212121',

      success: '#00D308',
      'success-hover': '#00b306',
```
this field of the config is for the `color` prop, which includes color types supported from our design system `default/normal/warning/error`, as well as the disabled/hovering effects of each variant (`filled/outlined/shaded/borderless`) for each color type
if none of these color types match what the user of these buttons is looking for, then they could (1) import the ButtonBase component & apply their own styles, or (2) change just the color via `className` (i'll open up options to override colors/borders/shadings/etc)